test/test_fx.py

```diff
@@ -476,6 +482,88 @@ def to_trace(y):
+    def test_wrap_with_make_fx(self):
+        def to_trace(y):
+            return a_lifted_leaf((4, y), 3) * a_lifted_leaf((3, 4), 5) * a_lifted_leaf((y, y), y)
+
+        expected_code = """def forward(self, y_1):
+    a_lifted_leaf = __main___a_lifted_leaf((4, y_1), 3)
+    a_lifted_leaf_1 = __main___a_lifted_leaf((3, 4), 5)
+    mul = torch.ops.aten.mul.Tensor(a_lifted_leaf, 12);  a_lifted_leaf = None
+    a_lifted_leaf_2 = __main___a_lifted_leaf((y_1, y_1), y_1);  y_1 = None
+    mul_1 = torch.ops.aten.mul.Tensor(mul, a_lifted_leaf_2);  mul = a_lifted_leaf_2 = None
+    return mul_1"""
+
+        m = make_fx(to_trace, tracing_mode="real")(torch.tensor([10]))
+        self.assertIn('a_lifted_leaf', m.code)
+        # aten.add.Tensor should be internal to `a_lifted_leaf` when some of the parameters are tensors.
+        # However, it should not be traced as the function is marked as opaque.
+        self.assertNotIn('aten.add.Tensor', m.code)
+        self.assertExpectedInline(
+            m.code.strip(),
+            expected_code
+        )
+
+        m = make_fx(to_trace, tracing_mode="fake")(torch.tensor([10]))
+        self.assertIn('a_lifted_leaf', m.code)
+        self.assertNotIn('aten.add.Tensor', m.code)
+        self.assertExpectedInline(
+            m.code.strip(),
+            expected_code
+        )
+
+        m = make_fx(to_trace, tracing_mode="symbolic")(torch.tensor([10]))
+        self.assertIn('a_lifted_leaf', m.code)
+        self.assertNotIn('aten.add.Tensor', m.code)
+        self.assertExpectedInline(
+            m.code.strip(),
+            expected_code
+        )
+
+    def test_make_fx_should_not_trace_getitem(self):
+        class MaskedAddition(torch.nn.Module):
+            def __init__(self, patch_bias=False) -> None:
+                super().__init__()
+                self.bias = torch.nn.Parameter(torch.randn(2, 2, 8, 8))
+                self.patch_bias = patch_bias
+
+            def forward(self, x):
+                # Code to mimic a mask generation in GPT-like models.
+                y = x + x
+                m = y.size(-2)
+                n = y.size(-1)
+                if self.patch_bias:
+                    # Wrap the indexing into self.bias in an opaque function. `fx.symbolic_trace``
+                    # and `make_fx`` should not trace into the function; that is,
+                    # a FX `call_function` node with `target=patched_bias` should
+                    # always exist.
+                    b = patched_bias(self.bias, m, n)
+                else:
+                    # This triggers "__getitem__".
+                    b = self.bias[:, :, 0:m, 0:n]
+                return y + b
+        try:
+            from torch.fx._symbolic_trace import _wrapped_methods_to_patch
+            _wrapped_methods_to_patch.append((torch.Tensor, "__getitem__"))
+            x = torch.randn(2, 2, 4, 4)
+            traced = torch.fx.symbolic_trace(MaskedAddition())
```

Comment:
`dynamo.export` doesn't seem work here. It will decompose. Do you think `dynamo.export` should be tested? @SherlockNoMad, any comment? I think supporting custom op is important in exporter.

Reply:
This is false. Export does record `getitem`:

```

opcode         name       target                       args                                                                                                   kwargs
-------------  ---------  ---------------------------  -----------------------------------------------------------------------------------------------------  --------
placeholder    arg0       arg0                         ()                                                                                                     {}
call_function  add        <built-in function add>      (arg0, arg0)                                                                                           {}
get_attr       self_bias  self_bias                    ()                                                                                                     {}
call_function  getitem    <built-in function getitem>  (self_bias, (slice(None, None, None), slice(None, None, None), slice(0, 4, None), slice(0, 4, None)))  {}
call_function  add_1      <built-in function add>      (add, getitem)                                                                                         {}
output         output     output                       ([add_1],)                                                                                             {}
.
----------------------------------------------------------------------
Ran 1 test in 0.172s

OK
```

Reply:
If I pass `aten_graph=True`, I do indeed get the decomposition, as expected:
```
opcode         name              target             args                                           kwargs
-------------  ----------------  -----------------  ---------------------------------------------  --------
placeholder    arg0              arg0               ()                                             {}
call_function  add_tensor        aten.add.Tensor    (arg0, arg0)                                   {}
get_attr       _param_constant0  _param_constant0   ()                                             {}
call_function  slice_tensor      aten.slice.Tensor  (_param_constant0, 0, 0, 9223372036854775807)  {}
call_function  slice_tensor_1    aten.slice.Tensor  (slice_tensor, 1, 0, 9223372036854775807)      {}
call_function  slice_tensor_2    aten.slice.Tensor  (slice_tensor_1, 2, 0, 4)                      {}
call_function  slice_tensor_3    aten.slice.Tensor  (slice_tensor_2, 3, 0, 4)                      {}
call_function  add_tensor_1      aten.add.Tensor    (add_tensor, slice_tensor_3)                   {}
output         output            output             ([add_tensor_1],)                              {}
.
----------------------------------------------------------------------
Ran 1 test in 0.169s

OK
```

Reply:
I see. In ONNX exporter, I use `aten_graph=True` to reduce the number of ops, so I should set `aten_graph=False` instead. @SherlockNoMad, am I right? If I want an Aten graph from dynamo exporter, I must not use custom function.

Reply:
> If I pass aten_graph=True, I do indeed get the decomposition, as expected:

Actually, is this expected? If a function is wrapped, how come I can still decompose it...? Hmm...

Reply:
Right, dynamo.export doesn't support Autograd.Function yet, but it's a valid request.

cc @yanboliang

Reply:
@SherlockNoMad, it's not `Autograd.Function`. It's `fx.wrap` function. Both are custom functions.